### PR TITLE
RTT logging, probe-run

### PR DESCRIPTION
**Pull request by jordens**
- use rtt logging
- remove rtt feature
- update hitl script
- README: streamline docs a bit, add probe-run
- CHANGELOG: update
- README: document lockin, cleanup instructions, use probe-run
- set target-cpu=cortex-m7, MSRV 1.52.0, use probe-run

### CODE REVIEW

**Comment by jordens** [comment]
bors try

**Comment by jordens** [comment]
bors try

**Comment by jordens** [comment]
bors try

Bors doesn't look very cooperative today. HITL results are at https://github.com/quartiq/stabilizer/runs/2574636512?check_suite_focus=true

File: .cargo/config [hunk 1]
```diff
@@ -1,15 +1,10 @@
 [target.'cfg(all(target_arch = "arm", target_os = "none"))']
-runner = "gdb-multiarch -q -x openocd.gdb"
+runner = "probe-run --chip STM32H743ZITx --speed 30000"
```
Comment (ryan-summers) [inline .cargo/config:2]: Does probe-run spin up a GDB server as well? Or is this just a command to program the chip?

Comment (ryan-summers) [reply .cargo/config:2]: In general, I'd prefer if we kept around an easy way to spin up GDB, which was previously probe-run. The command-line way to flash was previously just `cargo embed --bin dual-iir`, and `cargo run --bin dual-iir` was meant for interactive debug

Comment (jordens) [reply .cargo/config:2]: There are still two ways to flash and while still getting gdb access as before. Just the command to start gdb changes.

* Start `cargo embed` with options to provide the gdb stub (https://github.com/probe-rs/cargo-embed/blob/master/src/config/default.toml#L71) or start `openocd` as before.
* `cargo build --release --bin app && gdb-multiarch -q -x openocd.gdb path/to/app` to build and start gdb.

You can still use `cargo embed` to flash in the same way. That doesn't change. The command to start GDB becomes a bit longer.
The tooling for GDB+openocd is (a) significantly more complex to install (b) we don't test it, (c) doesn't provide usable logging, and (d) it isn't used as frequently compared to "just" flashing. Therefore I'd rather have a low entry barrier, streamlined runner that gives me backtraces and convenient logging for debugging.

Comment (ryan-summers) [reply .cargo/config:2]: Ah, probe-run gives a logging interface. That's no issue - I can get gdb set up locally or see if I can make a custom cargo profile or something.

**Review by ryan-summers — APPROVED** [review]

File: .cargo/config (hunk 1 above)
Comment (jordens) [reply .cargo/config:2]: A cargo profile might be nice for gdb.

**Comment by jordens** [comment]
bors merge

**Pull request merged**